kernel/environ.ml

```diff
@@ -405,6 +405,14 @@ let vars_of_global env constr =
     (** FIXME: is Proj missing? *)
     | _ -> raise Not_found
 
+let vars_of_global_reference env gr =
+  let open Globnames in
+  match gr with
+  | VarRef id -> Id.Set.singleton id
+  | ConstRef kn -> lookup_constant_variables kn env
+  | IndRef ind -> lookup_inductive_variables ind env
+  | ConstructRef cstr -> lookup_constructor_variables cstr env
+
```

Comment:
I would prefer to move global_reference to the kernel, but that sounds a bit late for 8.6, I can move this outside the kernel.